SF iOS/SF iOS/RemoteUIImage/UIImage+URL.m

```diff
@@ -18,15 +18,21 @@ + (void)fetchImageFromURL:(NSURL *)fileURL onQueue:(nullable NSOperationQueue *)
     }
 
     [queue addOperationWithBlock:^{
-        NSData *data = [NSData dataWithContentsOfURL:fileURL];
-        UIImage *image = nil;
-        if (data) {
-            image = [UIImage imageWithData:data];
-        }
+        NSURLSession *sharedSession = [NSURLSession sharedSession];
+        NSURLRequest *request = [[NSURLRequest alloc] initWithURL:fileURL
+                                                      cachePolicy:NSURLRequestReturnCacheDataElseLoad
+                                                  timeoutInterval:30];
         
-        [[NSOperationQueue mainQueue] addOperationWithBlock:^{
-            completionHandler(image, nil);
-        }];
+        [[sharedSession dataTaskWithRequest:request
+                         completionHandler:^(NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable error) {
+            UIImage *image = nil;
+            if (data) {
+                image = [UIImage imageWithData:data];
```

Comment:
I kind of wonder if the nil check is necessary for this `imageWithData`. My guess is that it'd return nil if you gave it nil

Reply:
I see that it was in the original implementation too. I can check this out on my own.